clientconn.go

```diff
@@ -491,6 +495,9 @@ func (cc *ClientConn) getTransport(ctx context.Context, opts BalancerGetOptions)
 		cc.mu.RLock()
 		if cc.conns == nil {
 			cc.mu.RUnlock()
+			if put != nil {
+				put()
```

Comment:
It's related to the connection path making sense, but not this particular commit. Reverted.